src/DDTrace/Log/AbstractLogger.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace DDTrace\Log;
+
+/**
+ * An abstract logger.
+ */
+abstract class AbstractLogger implements LoggerInterface
+{
+    /**
+     * @var array A map - 'name' => true|false - of enabled levels.
+     */
+    private $enabledLevels = [];
+
+    /**
+     * @param string $level
+     */
+    public function __construct($level)
+    {
+        // At the moment we only support debug or off.
+        foreach (LogLevel::all() as $knownLevel) {
```

Comment:
This looks to me a bit like overoptimization. Can multiple levels be active at the same time? 

Just asking since it looks like it required quite a bit of refactoring to make it work this way.

Reply:
Ok seems like I misunderstood intended use.

Reply:
I think if the method was implemented fully, I wouldn't misunderstood it so easily.

@labbati WDYT about implementation like this:

```php
# all() have all levels ordered from highest to lowest
# all preceding levels to given $level will be marked as enabled
foreach (LogLevel::all() as $knownLevel) {
    if (!$found) {
        if ($level === $knownLevel){
            $found = true;
        }
        $this->enabledLevels[$knownLevel] = true;
    } else {
        $this->enabledLevels[$knownLevel] = false;
    }
}
```

Reply:
Thanks @pawelchcki you are totally right, I think that implementing just what we needed this time was a real mistake 😄

Reply:
One simple note. If you don't mind I would invert the logic. With the code proposed here, if the user insert a wrong value (e.g. `my_level`) then I would have `debug` level activate while I would like to have logging off.

So something like

```
        # all() have all levels ordered from highest to lowest
        # all preceding levels to given $level will be marked as enabled
        $found = false;
        foreach (array_reverse(LogLevel::all()) as $knownLevel) {
            $found = $found || ($level === $knownLevel);
            $this->enabledLevels[$knownLevel] = $found || false;
        }
```

Reply:
Yea makes total sense!
Just one tidbit I would rename the `$found` to `$enabled`
Also Not sure if I understand this construct `$found || false` :D